### PR TITLE
initial set of rules for diversity and inclusion checks

### LINKED ISSUE #7: Update CI guideline tests to check inclusive language
Modify Vale and Lint config to match expectations from #6

### PR DESCRIPTION
**Pull request by NicolasMassart**
Add new inclusion and diversity rules to the Vale check.

it doesn't cover all possible cases but only the most obvious ones that we can find in our docs. This will have to be improved as we notice other non inclusive words in our projects.

Terms between quotes are regex, for example:`'black[ -]?list'` matches `black-list`,`blacklist`,`black list`

Note that these terms are not checked in code blocks, so any URLs including `master` like GitHub or CLI options like `--blacklist` won't be impacted. However, the description of these options will be checked. We could have for example:

```markdown
`--blacklist`: a list of nodes that can't connect to the network.
```

fixes #7

### CODE REVIEW

**Comment by shemnon** [comment]
1. How do we handle backwards compatability?  some whitelist and blacklist CLIs need to stay in place for compatability.
2. How do we deal with code referencing other github repos with "master" as the branch name?  It's in URLs typically but regexps will trigger it.  perhaps allow `/master/` but flag any other occurance?

File: build_tools/vale/vale_styles/Consensys/Inclusive.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+extends: substitution
+message: Consider using %s instead of '%s'
+link: https://github.com/Consensys/doc.common/wiki/Documentation-Style-Guide#5-be-inclusive
+ignorecase: true
+level: error
+swap:
+  bellow: following
```
Comment (macfarla) [inline build_tools/vale/vale_styles/Consensys/Inclusive.yml:7]: s/bellow/below

**Comment by macfarla** [comment]
Is this going to be an error or warning?

**Comment by NicolasMassart** [comment]
> Is this going to be an error or warning?

I think it should be an error otherwise it's not going to be taken in account. We don't even display warnings in the latest versions of this common module.
Note that this is not enforced in code blocks but only in regular text. Se even if you still have a CLI option called `--whitelist` somewhere, it's going to be fine if for instance it's describes as an allow list. We already started the conversion work on some of our projects and we have both names available but the main point is to be aware that these words are not expected to be used here anymore.
I'm still working on the list though. I raised this internally in a DEI channel and marked this PR as draft until I get enough feedback.

**Comment by NicolasMassart** [comment]
@macfarla after discussions, the rule is going to be a warning.
Note that I also fixed the group regex in the accept files... it's not a good practice to fix two issue sin the same PR, I know, I won't do that too often, promise.

File: build_tools/vale/vale_styles/Consensys/Inclusive.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+extends: substitution
+message: Consider using %s instead of "%s"
+link: https://github.com/Consensys/doc.common/wiki/Documentation-Style-Guide#5-be-inclusive
+ignorecase: true
+level: warning
+swap:
+  'black[ \-]?list(?:ing)?s?': '"denylist", "excludelist", or "blocklist"'
+  'white[ \-]?list(?:ing)?s?': '"allowlist", "trustlist", or "safelist"'
+  'gray[ \-]?list(?:ing)?s?': '"provisional" or "temporary" list'
```
Comment (macfarla) [inline build_tools/vale/vale_styles/Consensys/Inclusive.yml:9]: should we also include alternative spelling "grey"

File: build_tools/vale/vale_styles/Consensys/Inclusive.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  'masters?': '"main", "parent" or "primary"'
+  'slaves?': '"replica" or "subordinate"'
+  'm[ae]n': '"person"'
+  'natives?':  'a more precise term like "built-in"'
+  'first[ \-]?class citizens?': '"core feature" or "top-level"'
```
Comment (macfarla) [inline build_tools/vale/vale_styles/Consensys/Inclusive.yml:14]: what about second-class citizens

File: build_tools/vale/vale_styles/Consensys/Inclusive.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+extends: substitution
+message: Consider using %s instead of "%s"
+link: https://github.com/Consensys/doc.common/wiki/Documentation-Style-Guide#5-be-inclusive
+ignorecase: true
+level: warning
+swap:
+  'black[ \-]?list(?:ing)?s?': '"denylist", "excludelist", or "blocklist"'
```
Comment (macfarla) [inline build_tools/vale/vale_styles/Consensys/Inclusive.yml:7]: prob also include "ed" as a suffix

**Review by macfarla — APPROVED** [review]
LGTM - couple of suggestions

File: build_tools/vale/vale_styles/Consensys/Inclusive.yml (hunk 3 above)
Comment (NicolasMassart) [reply build_tools/vale/vale_styles/Consensys/Inclusive.yml:14]: How is it used in computers context?

Comment (macfarla) [reply build_tools/vale/vale_styles/Consensys/Inclusive.yml:14]: I can't think of an example in besu but eg in meetings, people not in the room "nobody is a second-class citizen"

Comment (NicolasMassart) [reply build_tools/vale/vale_styles/Consensys/Inclusive.yml:14]: Ok, so maybe we don't need it right now. I suggest we start with this small list and be very vigilant during reviews and we add new terms when required.

**Pull request merged**